main/src/addins/MonoDevelop.AspNetCore/MonoDevelop.AspNetCore/LaunchProfileProvider.cs

```diff
@@ -36,27 +38,42 @@ namespace MonoDevelop.AspNetCore
 {
 	internal class LaunchProfileProvider
 	{
-		DotNetProject project;
+		readonly string baseDirectory;
+		readonly string defaultNamespace;
 		readonly object fileLocker = new object ();
 		public IDictionary<string, JToken> GlobalSettings { get; private set; }
 		internal JObject ProfilesObject { get; private set; }
-		internal string launchSettingsJsonPath => Path.Combine (project.BaseDirectory, "Properties", "launchSettings.json");
+		public ConcurrentDictionary<string, LaunchProfileData> Profiles { get; set; }
```

Comment:
Why can we publicly set the Profiles dictionary? Should this be readonly?
```suggestion
		public ConcurrentDictionary<string, LaunchProfileData> Profiles { get; }
```